calculator_test.go

```diff
@@ -1,24 +1,136 @@
+		{"Adding one large and one small number", Add, 9, 123191023123, []float64{}, 123191023132},
+		{"Adding two zeros yields zero", Add, 0, 0, []float64{}, 0},
+		{"Adding a number with a fraction and a number without one yields a number with fraction", Add, 1.2, 333, []float64{}, 334.2},
+		{"Adding a negative number and a positive number", Add, -73, 99, []float64{}, 26},
+		{"Adding two negative numbers yields a negative number", Add, -1, -1, []float64{}, -2},
+		{"Adding a number to zero yields the same number", Add, -2222, 0, []float64{}, -2222},
+		{"Adding five very small numbers", Add, 0.0000005133, 0.00000312, []float64{0.000664, 0.0000532, 0.00001}, 0.0007308333},
+
+		{"Subtracting a positive number from itself yields zero", Subtract, 1, 1, []float64{}, 0},
+		{"Subtracting a negative number from itself yields zero", Subtract, -51, -51, []float64{}, 0},
+		{"Subtracting multiple positive numbers from a much larger positive number yields a positive number", Subtract, 4323, 12, []float64{11, 39}, 4261},
+		{"Subtracting a positive number from a smaller positive number yields a negative number", Subtract, 11.2, 89.8, []float64{}, -78.6},
+		{"Subtracting a negative number from a larger negative number yields a positive number", Subtract, -100, -111, []float64{}, 11},
+		{"Subtracting multiple positive numbers from zero yields a negative number", Subtract, 0, 6, []float64{6, 15}, -27},
+		{"Subtracting a negative number from zero yields a positive number", Subtract, 0, -22, []float64{}, 22},
+
+		{"Multiplying 1 by 1 always yields 1", Multiply, 1, 1, []float64{}, 1},
+		{"Multiplying a number by 1 any number of times yields that number", Multiply, 3999191, 1, []float64{1, 1, 1}, 3999191},
+		{"Multiplying a number by itself equals that number squared", Multiply, 5091, 5091, []float64{}, 25918281},
+		{"Multiplying two negative numbers yields a positive number", Multiply, -10, -10, []float64{}, 100},
+		{"Multiplying three negative number yields a negative number", Multiply, -5, -12, []float64{-67}, -4020},
+		{"Multiplying a positive number by a negative number yields a negative number", Multiply, 12, -1.5, []float64{}, -18},
+		{"Multiplying any number by zero yields zero", Multiply, 43, 0, []float64{}, 0},
+	}
+	for _, c := range testCases {
+		got := c.fn(c.a, c.b, c.c...)
+		if c.want != got {
+			t.Errorf("%s: want %f, got %f", c.desc, c.want, got)
+		}
 	}
 }
 
-// func TestSubtract(t *testing.T) {
-// 	t.Parallel()
-// 	var want float64 = 2
-// 	got := calculator.Subtract(4, 2)
-// 	if want != got {
-// 		t.Errorf("want %f, got %f", want, got)
-// 	}
-// }
+func TestDivide(t *testing.T) {
+	t.Parallel()
+	var testCases = []struct {
+		name        string
+		a           float64
+		b           float64
+		c           []float64
+		want        float64
+		errExpected bool
+	}{
+		{"Dividing a number by 1 yields the same number", 10, 1, []float64{}, 10, false},
```

Comment:
It doesn't take a very big struct before it becomes quite hard to read nameless literals like this. I would give the field names, which also implies that we should write the struct over multiple lines, doesn't it?